packages/story-editor/src/components/header/buttons/update.js

```diff
@@ -75,38 +82,53 @@ function UpdateButton({ hasUpdates = false, forceIsSaving = false }) {
   // then only if there are new changes or the story has meta-boxes – as these
   // can update without us knowing it.
   const isEnabled = !isSaving && !isUploading && (hasNewChanges || hasUpdates);
-  let text;
-  switch (status) {
-    case 'publish':
-    case 'private':
-      text = __('Update', 'web-stories');
-      break;
-    case 'future':
-      text = __('Schedule', 'web-stories');
-      break;
-    default:
-      text = __('Save draft', 'web-stories');
-      return (
-        <Tooltip title={text} hasTail>
-          <Button
-            variant={BUTTON_VARIANTS.SQUARE}
-            type={BUTTON_TYPES.QUATERNARY}
-            size={BUTTON_SIZES.SMALL}
-            onClick={() => saveStory({ status: 'draft' })}
-            disabled={!isEnabled}
-            aria-label={text}
-          >
-            <Icons.FloppyDisk />
-          </Button>
-        </Tooltip>
-      );
+
+  if ('pending' === status) {
+    return (
+      <Tooltip title={__('Save as pending', 'web-stories')} hasTail>
+        <Button
+          variant={BUTTON_VARIANTS.SQUARE}
+          type={BUTTON_TYPES.QUATERNARY}
+          size={BUTTON_SIZES.SMALL}
+          onClick={() => saveStory()}
+          disabled={!isEnabled}
+          aria-label={__('Save as pending', 'web-stories')}
+        >
+          <Icons.FloppyDisk />
+        </Button>
+      </Tooltip>
+    );
+  }
+
+  if ('draft' === status) {
+    return (
+      <Tooltip title={__('Save draft', 'web-stories')} hasTail>
+        <Button
+          variant={BUTTON_VARIANTS.SQUARE}
+          type={BUTTON_TYPES.QUATERNARY}
+          size={BUTTON_SIZES.SMALL}
+          onClick={() => saveStory({ status: 'draft' })}
+          disabled={!isEnabled}
+          aria-label={__('Save draft', 'web-stories')}
+        >
+          <Icons.FloppyDisk />
+        </Button>
+      </Tooltip>
+    );
```

Comment:
```suggestion
  if ('pending' === status || 'draft' === status) {
    const text = 'pending' === status ? __('Save as pending', 'web-stories') : __('Save draft', 'web-stories');
    const saveState = 'pending' === status ? {} : { status: 'draft' };
    return (
      <Tooltip title={text} hasTail>
        <Button
          variant={BUTTON_VARIANTS.SQUARE}
          type={BUTTON_TYPES.QUATERNARY}
          size={BUTTON_SIZES.SMALL}
          onClick={() => saveStory(saveState)}
          disabled={!isEnabled}
          aria-label={text}
        >
          <Icons.FloppyDisk />
        </Button>
      </Tooltip>
    );
```

This means a lot less repeated code.

Reply:
I'll just extract the component